### PR TITLE
Update Tuya Climate for new esphome PR

### PR DESCRIPTION
**Pull request by moriahmorgan**
## Description:

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#5159

## Checklist:

  - [x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/climate/tuya.rst [hunk 1]
```diff
@@ -11,23 +11,33 @@ Tuya climate requires a :doc:`/components/tuya` to be configured.
...
+- 2 represents the target temperature.
+- 3 represents the current temperature
+- 4 represents the active state (operating mode, such as heat/cool). This is a value that can range from 0-4 on this device. Each value is a different mode.
+- 5 represents the Fan Speed / Fan Mode. This is a value that can range from 0-4 on this device. Each value is a different speed.
+- 8 represents the Eco Mode preset.
+- 101 represents the Sleep Mode preset.
+- 103 represents the Auto-Clean feature. (use the :doc:`/components/switch/tuya` component to control this)
+- 104 represents the display toggle (on device) feature. (use the :doc:`/components/switch/tuya` component to control this)
+- 106 represents the Vertical Swing switch.
+- 107 represents the Horizontal Swing switch.
```
Comment (jesserockz) [inline components/climate/tuya.rst]: We dont really need to change this block, its just one example.

If you want to add an example for a specific device, it should go on https://devices.esphome.io

File: components/climate/tuya.rst [hunk 2]
```diff
@@ -37,10 +47,32 @@ Based on this, you can create the climate device as follows:
...
+        swing_mode:
+          vertical_datapoint: 106
+          horizontal_datapoint: 107
+        fan_mode:
+          datapoint: 5
+          auto_value: 0
+          low_value: 2
+          medium_value: 3
+          middle_value: 4
+          high_value: 1
```
Comment (jesserockz) [inline components/climate/tuya.rst]: Same as above, except for the breaking change parts

File: components/climate/tuya.rst [hunk 3]
```diff
@@ -50,25 +82,57 @@ Configuration variables:
...
 - All other options from :ref:`Climate <config-climate>`.
 
+**Deprecated:**
+
+    - **active_state_datapoint** (*Optional*, int): Moved under the ``active_state`` config block.
+    - **active_state_heating_value** (*Optional*, int): Moved under the ``active_state`` config block.
+    - **active_state_cooling_value** (*Optional*, int): Moved under the ``active_state`` config block.
+    - **eco_datapoint** (*Optional*, int): Moved under the ``preset`` and ``eco`` config blocks.
+    - **eco_temperature** (*Optional*, float): Moved under the ``preset`` and ``eco`` config blocks.
+
```
Comment (jesserockz) [inline components/climate/tuya.rst]: I will add this in the changelog, we dont tend to add this to docs here
```suggestion
```

**Review by jesserockz — DISMISSED** [review]

File: components/climate/tuya.rst (hunk 1 above)
Comment (moriahmorgan) [reply components/climate/tuya.rst]: Agreed, I was just "updating" what was there, and was considering it to be a bit out of date.

File: components/climate/tuya.rst (hunk 2 above)
Comment (moriahmorgan) [reply components/climate/tuya.rst]: I think I got it in the right format this time.

**Comment by moriahmorgan** [comment]
Do I need to do anything more for the documentation here?

File: components/climate/tuya.rst (hunk 1 above)
Comment (jesserockz) [reply components/climate/tuya.rst]: I didnt mean delete the whole block... 
I just meant dont change it so much...haha

Comment (moriahmorgan) [reply components/climate/tuya.rst]: Ah ok, I will fix this. sorry for the delay. I have been pretty tied up lately.

**Comment by moriahmorgan** [comment]
Sorry for letting this sit. I will attempt to update tomorrow with the request you made. I have had a very busy last few months.

**Comment by moriahmorgan** [comment]
Everything should be ready for a code review, let me know if anything else needs to be done.

File: components/climate/tuya.rst [hunk 4]
```diff
@@ -50,22 +52,46 @@ Configuration variables:
 - **supports_heat** (*Optional*, boolean): Specifies if the device has a heating mode. Defaults to ``true``.
 - **supports_cool** (*Optional*, boolean): Specifies if the device has a cooling mode. Defaults to ``false``.
 - **switch_datapoint** (**Required**, int): The datapoint id number of the climate switch (device on/off).
-- **active_state_datapoint** (*Optional*, int): The datapoint id number of the active state - :ref:`see below <active_state_detection>`.
-- **active_state_heating_value** (*Optional*, int): The active state datapoint value the device reports when heating. Defaults to ``1`` - :ref:`see below <active_state_detection>`.
-- **active_state_cooling_value** (*Optional*, int): The active state datapoint value the device reports when cooling - :ref:`see below <active_state_detection>`.
+- **active_state** (*Optional*): Information for the Active State Configuration.
```
Comment (kbx81) [inline components/climate/tuya.rst]:
```suggestion
- **active_state** (*Optional*): Configuration for the Active State Configuration.
```

File: components/climate/tuya.rst [hunk 5]
```diff
@@ -50,22 +52,46 @@ Configuration variables:
...
+
+    - **datapoint** (**Required**, int): The datapoint id number of the active state - :ref:`see below <active_state_detection>`.
+    - **heating_value** (*Optional*, int): The active state datapoint value the device reports when heating. Defaults to ``1`` - :ref:`see below <active_state_detection>`.
+    - **cooling_value** (*Optional*, int): The active state datapoint value the device reports when cooling - :ref:`see below <active_state_detection>`.
+    - **drying_value** (*Optional*, int): The active state datapoint value the device reports when in drying mode.
+    - **fanonly_value** (*Optional*, int): The active state datapoint value the device reports when in Fan Only mode.
+- **preset** (*Optional*): Information for presets.
```
Comment (kbx81) [inline components/climate/tuya.rst]:
```suggestion
- **preset** (*Optional*): Configuration for presets.
```

File: components/climate/tuya.rst [hunk 6]
```diff
@@ -50,22 +52,46 @@ Configuration variables:
...
+
+    - **eco** (*Optional*): Config block for Eco preset.
```
Comment (kbx81) [inline components/climate/tuya.rst]:
```suggestion
    - **eco** (*Optional*): Configuration for Eco preset.
```

File: components/climate/tuya.rst [hunk 7]
```diff
@@ -50,22 +52,46 @@ Configuration variables:
...
+
+        - **datapoint** (**Required**, int): The datapoint id number of the Eco action.
+        - **temperature** (*Optional*, int): Temperature setpoint for Eco preset.
+    - **sleep** (*Optional*): Config block for Sleep preset
```
Comment (kbx81) [inline components/climate/tuya.rst]:
```suggestion
    - **sleep** (*Optional*): Configuration for Sleep preset
```

File: components/climate/tuya.rst [hunk 8]
```diff
@@ -50,22 +52,46 @@ Configuration variables:
...
+
+        - **datapoint** (**Required**, int): The Datapoint id number of the Sleep Action
+- **swing_mode** (*Optional*): Information for the Swing (Oscillation) modes.
```
Comment (kbx81) [inline components/climate/tuya.rst]:
```suggestion
- **swing_mode** (*Optional*): Configuration for the swing (oscillation) modes.
```

File: components/climate/tuya.rst [hunk 9]
```diff
@@ -50,22 +52,46 @@ Configuration variables:
...
+
+    - **vertical_datapoint** (*Optional*, int): The datapoint id number of the vertical swing action.
+    - **horizontal_datapoint** (*Optional*, int): The datapoint id number of the horizontal swing action.
+- **fan_mode** (*Optional*): Information for Fan Mode / Fan Speeds.
```
Comment (kbx81) [inline components/climate/tuya.rst]:
```suggestion
- **fan_mode** (*Optional*): Configuration for fan modes/fan speeds.
```

**Review by kbx81 — CHANGES_REQUESTED** [review]

**Review by kbx81 — APPROVED** [review]

**Pull request merged**